ci/deploy.sh

```diff
@@ -0,0 +1,122 @@
+
+set -ex
+
+# The deploy suffix flexibility is mainly here to allow
+# us to test Circle and Travis builds simultaneously on
+# the https://github.com/Automattic/vip-go-skeleton/ repo.
+DEPLOY_SUFFIX="${VIP_DEPLOY_SUFFIX:--built}"
+
+BRANCH="${CIRCLE_BRANCH:-$TRAVIS_BRANCH}"
+
+SRC_DIR="${TRAVIS_BUILD_DIR:-$PWD}"
+BUILD_DIR="/tmp/vip-go-build-$(date +%s)"
+
+if [[ $CIRCLECI ]]; then
+	CIRCLE_REPO_SLUG="${CIRCLE_PROJECT_USERNAME}/${CIRCLE_PROJECT_REPONAME}";
+fi
+REPO_SLUG=${CIRCLE_REPO_SLUG:-$TRAVIS_REPO_SLUG}
+REPO_SSH_URL="git@github.com:${REPO_SLUG}"
+COMMIT_SHA=${CIRCLE_SHA1:-$TRAVIS_COMMIT}
+DEPLOY_BRANCH="${BRANCH}${DEPLOY_SUFFIX}"
+cd $SRC_DIR
+COMMIT_AUTHOR_NAME="$( git log --format=%an -n 1 ${COMMIT_SHA} )"
+COMMIT_AUTHOR_EMAIL="$( git log --format=%ae -n 1 ${COMMIT_SHA} )"
+COMMIT_COMMITTER_NAME="$( git log --format=%cn -n 1 ${COMMIT_SHA} )"
+COMMIT_COMMITTER_EMAIL="$( git log --format=%ce -n 1 ${COMMIT_SHA} )"
+
+
+# Run some checks
+# ---------------
+
+if [[ -n $TRAVIS ]] && [[ -n $CIRCLECI ]]; then
+	echo "ERROR: this script requires either Circle CI or Travis CI"
+	echo "You will need to amend the setup above to set the required variables from the information specific to your CI service or tool."
+	exit 1
+fi
+
+if [[ -z "${BRANCH}" ]]; then
+	echo "ERROR: No branch specified!"
+	echo "This variable should be set by Travis CI and Circle CI, if you consistently experience errors please check with WordPress.com VIP support."
+	exit 1
+fi
+
+if [[ -d "$BUILD_DIR" ]]; then
+	echo "ERROR: ${BUILD_DIR} already exists."
+	echo "This should not happen, if you consistently experience errors please check with WordPress.com VIP support."
+	exit 1
+fi
+
+if [[ "${BRANCH}" == *${DEPLOY_SUFFIX} ]]; then
+	echo "NOTICE: Attempting to build from branch '${BRANCH}' to deploy '${DEPLOY_BRANCH}', seems like recursion so aborting."
+	echo "This is a protective measure, no action is required."
+	exit 0
+fi
+
+if [[ -n $TRAVIS ]] && [ $TRAVIS_PULL_REQUEST != 'false' ]; then
+	echo "NOTICE: Aborting a build to '${DEPLOY_BRANCH}' from a pull request on '${BRANCH}', only build from merges directly to the branch"
+	echo "This is a protective measure, no action is required."
+	exit 0
+fi
+
+# Everything seems OK, getting the built repo sorted
+# --------------------------------------------------
+
+echo "Deploying ${BRANCH} to ${DEPLOY_BRANCH}"
+
+# Making the directory we're going to sync the build into
+git init "${BUILD_DIR}"
+cd "${BUILD_DIR}"
+git remote add origin "${REPO_SSH_URL}"
+if [[ 0 = $(git ls-remote --heads "${REPO_SSH_URL}" "${DEPLOY_BRANCH}" | wc -l) ]]; then
+	echo -e "\nCreating a ${DEPLOY_BRANCH} branch..."
+	git checkout --quiet --orphan "${DEPLOY_BRANCH}"
+else
+	echo "Using existing ${DEPLOY_BRANCH} branch"
+	git fetch origin "${DEPLOY_BRANCH}"
+	git checkout --quiet "${DEPLOY_BRANCH}"
+fi
+
+# Copy the files over
+# -------------------
+
+if ! command -v 'rsync'; then
+	sudo apt-get install -q -y rsync
```

Comment:
We should run `apt-get update` to avoid issues with out-of-date dependencies/versions of `rsync` and other tools we're using. (This came up on a client site today.)